test/xdg_desktop_portal_test.dart

```diff
@@ -2164,6 +2280,75 @@ void main() {
         equals(['http://localhost:1234']));
   });
 
+  test('screen cast', () async {
+    var server = DBusServer();
+    var clientAddress =
+        await server.listenAddress(DBusAddress.unix(dir: Directory.systemTemp));
+    addTearDown(() async {
+      await server.close();
+    });
+    final screenCastStreamsList = [
+      ScreenCastStream(
+        nodeId: 1,
+        id: '123',
+        sourceType: ScreenCastAvailableSourceType.monitor,
+        x: 1920,
+        y: 0,
+        width: 1920,
+        height: 1080,
+      ),
+      ScreenCastStream(
+        nodeId: 2,
+        id: '456',
+        sourceType: ScreenCastAvailableSourceType.monitor,
+        x: 1920,
+        y: 0,
+        width: 1920,
+        height: 1080,
+      ),
+    ];
+
+    var portalServer = MockPortalServer(
+      clientAddress,
+      screenCastSessionHandle:
+          '/org/freedesktop/portal/desktop/session/123/dart123',
```

Comment:
Something is not quite right here - this should be generated in MockPortalServer.